cli/picker/picker.go

```diff
@@ -21,6 +21,19 @@ func HandlePicker(args []string) []string {
 	// If the command is build, test, or query without a specified target - apply to all targets.
 	command := arg.GetCommand(args)
 
+	bzlArgs := arg.GetBazelArgs(args)
+	for _, bzlArg := range bzlArgs {
+		// Skip using the picker if the user has specified a query file.
+		if strings.Contains(command, "query") && strings.Contains(bzlArg, "--query_file") {
+			return args
+		}
+
+		// Skip using the picker if the user has specified a target pattern file.
+		if command == "build" && strings.Contains(bzlArg, "--target_pattern_file") {
+			return args
+		}
```

Comment:
sorry I misread this originally - instead of Contains, should do `==` OR `startswith --target_pattern_file=`

Reply:
I think startswith makes sense. The flag could be set as `--target_pattern_file=targets.txt` OR `--target_pattern_file targets.txt --` (this happens before `--` so targets.txt won't be parsed as build target pattern).

Let me send a fix.